{
	"alias": "FEDORA-2024-1230cb2cd6",
	"autokarma": true,
	"autotime": true,
	"bugs": [
		{
			"bug_id": 2274782,
			"security": true,
			"title": "TRIAGE CVE-2024-3651 mingw-python-idna: python-idna: potential DoS via resource consumption via specially crafted inputs to idna.encode() [fedora-all]",
			"bugzilla": {
				"bug_id": "2274782",
				"creation_ts": "2024-04-12 19:23:30 +0000",
				"short_desc": "TRIAGE CVE-2024-3651 mingw-python-idna: python-idna: potential DoS via resource consumption via specially crafted inputs to idna.encode() [fedora-all]",
				"delta_ts": "2024-04-15 02:28:01 +0000",
				"bug_status": "ON_QA",
				"keywords": "Security, SecurityTracking",
				"priority": "medium",
				"bug_severity": "medium",
				"blocked": [
					{
						"bug_id": "2274779",
						"alias": "CVE-2024-3651",
						"creation_ts": "2024-04-12 19:20:16 +0000",
						"short_desc": "CVE-2024-3651 python-idna: potential DoS via resource consumption via specially crafted inputs to idna.encode()",
						"delta_ts": "2024-04-18 13:45:17 +0000",
						"bug_status": "NEW",
						"keywords": "Security",
						"priority": "medium",
						"bug_severity": "medium",
						"depends_on": [
							"2274780",
							"2274781",
							"2274782",
							"2274783",
							"2274784",
							"2274785",
							"2274786",
							"2274787",
							"2274788",
							"2274789",
							"2274790"
						],
						"blocked": [
							{
								"bug_id": "2274791",
								"error": "NotPermitted"
							}
						],
						"long_desc": [
							{
								"isprivate": "0",
								"commentid": "17969525",
								"comment_count": "0",
								"who": {
									"text": "mbenatto",
									"name": "Marco Benatto"
								},
								"bug_when": "2024-04-12 19:20:16 +0000",
								"thetext": "A specially crafted argument to the idna.encode() function could consume significant resources. This may lead to a denial-of-service.\n\nReference:\nhttps://github.com/kjd/idna/security/advisories/GHSA-jjg7-2v4v-x38h"
							},
							{
								"isprivate": "0",
								"commentid": "17969535",
								"comment_count": "1",
								"who": {
									"text": "mbenatto",
									"name": "Marco Benatto"
								},
								"bug_when": "2024-04-12 19:23:46 +0000",
								"thetext": "Created mingw-python-idna tracking bugs for this issue:\n\nAffects: fedora-all [bug 2274782]\n\n\nCreated python-idna tracking bugs for this issue:\n\nAffects: fedora-all [bug 2274780]\n\n\nCreated python-idna-ssl tracking bugs for this issue:\n\nAffects: epel-8 [bug 2274783]\nAffects: fedora-all [bug 2274781]"
							},
							{
								"isprivate": "0",
								"commentid": "17972058",
								"comment_count": "3",
								"who": {
									"text": "lbalhar",
									"name": "Lumír Balhar"
								},
								"bug_when": "2024-04-15 14:35:16 +0000",
								"thetext": "I don't know how to reproduce the issue. Even the changelog of idna does not mention it, the fix seems to be this commit: https://github.com/kjd/idna/commit/5beb28b9dd77912c0dd656d8b0fdba3eb80222e7\n\nI'm able to create something like this:\n\n```\nzwnj = '\\u200c'\nlatin = '\\u0061'\n\nidna.encode(latin * 10 + zwnj)\n```\n\nWith that input, the first for loop in the valid_contextj function runs 10 times (v3.6) instead of just once (v3.7). However, I'm not able to prepare an input where a significant difference between 3.6 and 3.7 would be visible when it comes to consumed resources or processing time.\n\nSo far, I'm only able to reproduce and verify the issue using cProfiler from Python. The following line:\n\nidna.encode(latin * 1000 + zwnj)\n\nProduces the following output of cProfiler for idna 3.6:\n\n$ python3 -m cProfile -s ncalls poc.py | head\n\n         14231 function calls (14181 primitive calls) in 0.005 seconds\n\n   Ordered by: call count\n\n   ncalls  tottime  percall  cumtime  percall filename:lineno(function)\n     5007    0.000    0.000    0.000    0.000 {built-in method builtins.ord}\n     2003    0.000    0.000    0.000    0.000 intranges.py:35(_decode_range)\n1096/1095    0.000    0.000    0.000    0.000 {built-in method builtins.len}\n     1024    0.000    0.000    0.000    0.000 {method 'get' of 'dict' objects}\n     1002    0.001    0.000    0.001    0.000 intranges.py:39(intranges_contain)\n\nand for idna 3.7:\n\n$ python3 -m cProfile -s ncalls poc.py | head\n\n         9337 function calls (9284 primitive calls) in 0.018 seconds\n\n   Ordered by: call count\n\n   ncalls  tottime  percall  cumtime  percall filename:lineno(function)\n     2003    0.000    0.000    0.000    0.000 intranges.py:35(_decode_range)\n1096/1095    0.000    0.000    0.000    0.000 {built-in method builtins.len}\n     1011    0.000    0.000    0.000    0.000 {built-in method builtins.ord}\n     1002    0.001    0.000    0.001    0.000 intranges.py:39(intranges_contain)\n     1002    0.000    0.000    0.000    0.000 intranges.py:32(_encode_range)\n\nSee the difference in the total function calls caused mostly by 5007 calls to ord function in the vulnerable version."
							},
							{
								"isprivate": "0",
								"commentid": "17977105",
								"comment_count": "4",
								"who": {
									"text": "lbalhar",
									"name": "Lumír Balhar"
								},
								"bug_when": "2024-04-18 08:36:40 +0000",
								"thetext": "Reproducer has been provided in my issue: https://github.com/kjd/idna/issues/175"
							}
						]
					}
				],
				"long_desc": [
					{
						"isprivate": "0",
						"commentid": "17969531",
						"comment_count": "0",
						"who": {
							"text": "mbenatto",
							"name": "Marco Benatto"
						},
						"bug_when": "2024-04-12 19:23:30 +0000",
						"thetext": "\nMore information about this security flaw is available in the following bug:\n\nhttp://bugzilla.redhat.com/show_bug.cgi?id=2274779\n\nDisclaimer: Community trackers are created by Red Hat Product Security team on a best effort basis. Package maintainers are required to ascertain if the flaw indeed affects their package, before starting the update process."
					},
					{
						"isprivate": "0",
						"commentid": "17969532",
						"comment_count": "1",
						"who": {
							"text": "mbenatto",
							"name": "Marco Benatto"
						},
						"bug_when": "2024-04-12 19:23:31 +0000",
						"thetext": "Use the following template to for the 'fedpkg update' request to submit an\nupdate for this issue as it contains the top-level parent bug(s) as well as\nthis tracking bug.  This will ensure that all associated bugs get updated\nwhen new packages are pushed to stable.\n\n=====\n\n# bugfix, security, enhancement, newpackage (required)\ntype=security\n\n# low, medium, high, urgent (required)\nseverity=medium\n\n# testing, stable\nrequest=testing\n\n# Bug numbers: 1234,9876\nbugs=2274779,2274782\n\n# Description of your update\nnotes=Security fix for [PUT CVEs HERE]\n\n# Enable request automation based on the stable/unstable karma thresholds\nautokarma=True\nstable_karma=3\nunstable_karma=-3\n\n# Automatically close bugs when this marked as stable\nclose_bugs=True\n\n# Suggest that users restart after update\nsuggest_reboot=False\n\n======\n\nAdditionally, you may opt to use the bodhi web interface to submit updates:\n\nhttps://bodhi.fedoraproject.org/updates/new"
					},
					{
						"isprivate": "0",
						"commentid": "17971298",
						"comment_count": "2",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-14 21:35:54 +0000",
						"thetext": "FEDORA-2024-831b7c8340 (mingw-python-idna-3.7-1.fc38) has been submitted as an update to Fedora 38.\nhttps://bodhi.fedoraproject.org/updates/FEDORA-2024-831b7c8340"
					},
					{
						"isprivate": "0",
						"commentid": "17971299",
						"comment_count": "3",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-14 21:35:54 +0000",
						"thetext": "FEDORA-2024-83ef5f3c4f (mingw-python-idna-3.7-1.fc39) has been submitted as an update to Fedora 39.\nhttps://bodhi.fedoraproject.org/updates/FEDORA-2024-83ef5f3c4f"
					},
					{
						"isprivate": "0",
						"commentid": "17971300",
						"comment_count": "4",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-14 21:35:56 +0000",
						"thetext": "FEDORA-2024-1230cb2cd6 (mingw-python-idna-3.7-1.fc40) has been submitted as an update to Fedora 40.\nhttps://bodhi.fedoraproject.org/updates/FEDORA-2024-1230cb2cd6"
					},
					{
						"isprivate": "0",
						"commentid": "17971379",
						"comment_count": "5",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-15 01:12:01 +0000",
						"thetext": "FEDORA-2024-83ef5f3c4f has been pushed to the Fedora 39 testing repository.\nSoon you'll be able to install the update with the following command:\n`sudo dnf upgrade --enablerepo=updates-testing --refresh --advisory=FEDORA-2024-83ef5f3c4f`\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2024-83ef5f3c4f\n\nSee also https://fedoraproject.org/wiki/QA:Updates_Testing for more information on how to test updates."
					},
					{
						"isprivate": "0",
						"commentid": "17971440",
						"comment_count": "6",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-15 02:19:23 +0000",
						"thetext": "FEDORA-2024-1230cb2cd6 has been pushed to the Fedora 40 testing repository.\nSoon you'll be able to install the update with the following command:\n`sudo dnf upgrade --enablerepo=updates-testing --refresh --advisory=FEDORA-2024-1230cb2cd6`\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2024-1230cb2cd6\n\nSee also https://fedoraproject.org/wiki/QA:Updates_Testing for more information on how to test updates."
					},
					{
						"isprivate": "0",
						"commentid": "17971445",
						"comment_count": "7",
						"who": {
							"text": "updates",
							"name": "Fedora Update System"
						},
						"bug_when": "2024-04-15 02:28:01 +0000",
						"thetext": "FEDORA-2024-831b7c8340 has been pushed to the Fedora 38 testing repository.\nSoon you'll be able to install the update with the following command:\n`sudo dnf upgrade --enablerepo=updates-testing --refresh --advisory=FEDORA-2024-831b7c8340`\nYou can provide feedback for this update here: https://bodhi.fedoraproject.org/updates/FEDORA-2024-831b7c8340\n\nSee also https://fedoraproject.org/wiki/QA:Updates_Testing for more information on how to test updates."
					}
				]
			}
		}
	],
	"builds": [
		{
			"epoch": 0,
			"nvr": "mingw-python-idna-3.7-1.fc40",
			"release_id": 76,
			"signed": true,
			"type": "rpm",
			"package": {
				"noarch": [
					{
						"name": "mingw32-python3-idna",
						"version": "3.7",
						"release": "1.fc40",
						"arch": "noarch"
					},
					{
						"name": "mingw64-python3-idna",
						"version": "3.7",
						"release": "1.fc40",
						"arch": "noarch"
					}
				],
				"src": [
					{
						"name": "mingw-python-idna",
						"version": "3.7",
						"release": "1.fc40",
						"arch": "src"
					}
				]
			}
		}
	],
	"close_bugs": true,
	"comments": [
		{
			"id": 3490287,
			"text": "This update has been submitted for testing by smani. ",
			"timestamp": "2024-04-14 21:35:53",
			"update_id": 604688,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 3490288,
			"text": "This update's test gating status has been changed to 'ignored'.",
			"timestamp": "2024-04-14 21:35:54",
			"update_id": 604688,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		},
		{
			"id": 3490552,
			"text": "This update has been pushed to testing.",
			"timestamp": "2024-04-15 02:19:33",
			"update_id": 604688,
			"user": {
				"avatar": "https://apps.fedoraproject.org/img/icons/bodhi-24.png",
				"id": 91,
				"name": "bodhi",
				"openid": "bodhi.id.fedoraproject.org"
			},
			"user_id": 91
		}
	],
	"content_type": "rpm",
	"critpath_groups": "",
	"date_pushed": "2024-04-15 02:18:35",
	"date_submitted": "2024-04-14 21:35:53",
	"date_testing": "2024-04-15 02:18:35",
	"notes": "Update to idna-3.7.",
	"pushed": true,
	"release": {
		"branch": "f40",
		"candidate_tag": "f40-updates-candidate",
		"composed_by_bodhi": true,
		"create_automatic_updates": false,
		"dist_tag": "f40",
		"id_prefix": "FEDORA",
		"long_name": "Fedora 40",
		"mail_template": "fedora_errata_template",
		"name": "F40",
		"override_tag": "f40-override",
		"package_manager": "dnf",
		"pending_signing_tag": "f40-signing-pending",
		"pending_stable_tag": "f40-updates-pending",
		"pending_testing_tag": "f40-updates-testing-pending",
		"stable_tag": "f40",
		"state": "frozen",
		"testing_repository": "updates-testing",
		"testing_tag": "f40-updates-testing",
		"version": "40"
	},
	"require_bugs": true,
	"require_testcases": true,
	"severity": "medium",
	"stable_days": 7,
	"stable_karma": 3,
	"status": "testing",
	"suggest": "unspecified",
	"test_gating_status": "ignored",
	"title": "mingw-python-idna-3.7-1.fc40",
	"type": "security",
	"url": "https://bodhi.fedoraproject.org/updates/FEDORA-2024-1230cb2cd6",
	"unstable_karma": -3,
	"updateid": "FEDORA-2024-1230cb2cd6",
	"user": {
		"avatar": "https://seccdn.libravatar.org/avatar/45d3b000186539c7dfca915d2fa0176ec9db448e552c9cf548251f4922e664bd?s=24&d=retro",
		"email": "manisandro@gmail.com",
		"groups": [
			{
				"name": "packager"
			},
			{
				"name": "provenpackager"
			},
			{
				"name": "ipausers"
			},
			{
				"name": "fedora-contributor"
			},
			{
				"name": "signed_fpca"
			},
			{
				"name": "fedorabugs"
			}
		],
		"id": 117,
		"name": "smani",
		"openid": "smani.id.fedoraproject.org"
	},
	"version_hash": "9ff14cee7d76d86d9d29dd6ba096538f1a417fb2"
}
